File: static/code/sample/tethered/gdl-ldledr16-b.cs
// In this sample:
// Scan from led D1 to led D16

using GHIElectronics.DUELink;
using System;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

void SetLed(int index, bool value) {
    var val = value ? 1 : 0;
    duelink.Engine.ExecuteCommand($"SetLed({index},{val})");
}

var count = 0;
while (true) {
    SetLed((count % 16) + 1, (count / 16) % 2 == 0 ? true : false);
    count++;

    Thread.Sleep(100);
}

File: static/code/sample/tethered/gdl-ldledr16-b.js
// In this sample:
// Scan from led D1 to led D16

import pkg_serialusb from 'dlserialusb';
const {SerialUSB} = pkg_serialusb

import pkg_duelink from 'duelink';
const {DUELinkController} = pkg_duelink

let duelink = new DUELinkController(new SerialUSB());
await duelink.Connect();

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

async function SetLed(index, value) {
  const val = value ? 1 : 0;
  await duelink.Engine.ExecuteCommand(`SetLed(${index},${val})`);
}

async function main() {
  let count = 0;

  while (true) {
    await SetLed((count % 16) + 1, Math.floor(count / 16) % 2 === 0);
    count++;
    await sleep(100); // 100 ms
  }
}

main();

